src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -63,7 +68,9 @@ public class LanguageServer
         /// </summary>
         public NLHandler NL2FxImplementation { get; set; }
```

Comment:
Keeping this for backward compat reasons and using it as a backup to factory. If i remove this, then it might introduce compile time error in other consumers of Language Server.

Reply:
Given we only create one SDK instance, Nl2FxImplementation property being overwritten from the outside is not a thread safe. The factory approach makes the create of NlHandler dynamic and scopes it the IPowerFxInstance which is different for any language server task so we should be thread safe